src/topp/MSGFPlusAdapter.cpp

```diff
@@ -388,6 +388,31 @@ class MSGFPlusAdapter :
     }
   }
   
+  String describeHit_(const PeptideHit& hit)
+  {
+    return "peptide hit with sequence '" + hit.getSequence().toString() +
+      "', charge " + String(hit.getCharge()) + ", score " + 
+      String(hit.getScore());
+  }
+
+  // Set the MS-GF+ e-value (MS:1002052) as new peptide identification score.
+  void switchScores_(PeptideIdentification& id)
+  {
+    for (vector<PeptideIdentification::HitType>::iterator hit_it = id.getHits().begin(); hit_it != id.getHits().end(); ++hit_it)
+    {
+      // MS:1002052 == MS-GF spectral E-value
+      if (!hit_it->metaValueExists("MS:1002052"))
+      {
+        String msg = "Meta value 'MS:1002052' not found for " + describeHit_(*hit_it);
+        throw Exception::MissingInformation(__FILE__, __LINE__, __PRETTY_FUNCTION__, msg);
+      }
+      
+      hit_it->setScore(hit_it->getMetaValue("MS:1002052"));
+    }
+    id.setScoreType("MSGFPlus");
```

Comment:
I think the score type should be "SpecEValue", as in line 681 (old code)/706 (your code).